packages/cli/commands/import/workflow.ts

```diff
@@ -94,40 +105,83 @@ export class ImportWorkflowsCommand extends Command {
 					if (credentialsEntities.length > 0) {
-						// eslint-disable-next-line
 						workflow.nodes.forEach((node: INode) => {
 							this.transformCredentials(node, credentialsEntities);
 						});
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
 					await Db.collections.Workflow!.save(workflow);
+
+					const sharedWorkflow = new SharedWorkflow();
+
+					await Db.collections.SharedWorkflow!.save(
+						Object.assign(sharedWorkflow, {
+							user: owner,
+							workflow,
+							role: ownerWorkflowRole,
+						}),
+					);
 				}
 			} else {
-				const fileContents = JSON.parse(fs.readFileSync(flags.input, { encoding: 'utf8' }));
+				const workflows = JSON.parse(fs.readFileSync(flags.input, { encoding: 'utf8' }));
 
-				if (!Array.isArray(fileContents)) {
-					throw new Error(`File does not seem to contain workflows.`);
+				if (!Array.isArray(workflows)) {
+					throw new Error(
+						'File does not seem to contain workflows. Make sure the workflows are contained in an array.',
+					);
 				}
 
-				for (i = 0; i < fileContents.length; i++) {
+				for (i = 0; i < workflows.length; i++) {
 					if (credentialsEntities.length > 0) {
-						// eslint-disable-next-line
-						fileContents[i].nodes.forEach((node: INode) => {
+						workflows[i].nodes.forEach((node: INode) => {
 							this.transformCredentials(node, credentialsEntities);
 						});
 					}
-					// eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-non-null-assertion
-					await Db.collections.Workflow!.save(fileContents[i]);
+					await Db.collections.Workflow!.save(workflows[i]);
+
+					const sharedWorkflow = new SharedWorkflow();
+
+					await Db.collections.SharedWorkflow!.save(
+						Object.assign(sharedWorkflow, {
+							user: owner,
+							workflow: workflows[i],
+							role: ownerWorkflowRole,
+						}),
+					);
 				}
 			}
 
 			console.info(`Successfully imported ${i} ${i === 1 ? 'workflow.' : 'workflows.'}`);
-			process.exit(0);
+			process.exit();
 		} catch (error) {
-			console.error('An error occurred while exporting workflows. See log messages for details.');
-			// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-			logger.error(error.message);
+			console.error('An error occurred while importing workflows. See log messages for details.');
+			if (error instanceof Error) logger.error(error.message);
 			this.exit(1);
 		}
 	}
+
+	private async getInstanceOwner(): Promise<User> {
+		const globalRole = await Db.collections.Role!.findOneOrFail({
+			name: 'owner',
+			scope: 'global',
+		});
+
+		const owner = await Db.collections.User!.findOne({ globalRole });
+
+		if (owner) return owner;
+
+		const user = new User();
+
+		await Db.collections.User!.save(
+			Object.assign(user, {
+				firstName: 'default',
+				lastName: 'default',
+				email: null,
+				password: null,
+				resetPasswordToken: null,
+				...globalRole,
+			}),
+		);
```

Comment:
reference other CLI command about creating the owner in a get function --> I think we should not create an owner in a import function because the DB is in a broken state and we should not attempt to clear it with an import command